pkg/lib/indexer/indexer.go

```diff
@@ -570,19 +586,83 @@ func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
 }
 
 type bundleDirPrefix struct {
-	pkgName, bundleVersion string
+	pkgName, name, bundleVersion, replaces string
+	skips                                  map[string]struct{}
 }
 
 func getBundlesToExport(dbQuerier pregistry.Query, packages []string) (map[string]bundleDirPrefix, error) {
 	bundleMap := make(map[string]bundleDirPrefix)
-
+	graphLoader := &sqlite.SQLGraphLoader{
+		Querier: dbQuerier,
+	}
 	for _, packageName := range packages {
-		bundlesForPackage, err := dbQuerier.GetBundlesForPackage(context.TODO(), packageName)
+		bundleGraph, err := graphLoader.Generate(packageName)
 		if err != nil {
 			return nil, err
 		}
-		for k, _ := range bundlesForPackage {
-			bundleMap[k.BundlePath] = bundleDirPrefix{pkgName: packageName, bundleVersion: k.Version}
+		versionMap := make(map[string]string)
+		for _, channel := range bundleGraph.Channels {
+			var replaceDepth map[string]int64
+			for node, replaceNodes := range channel.Nodes {
+				var replaces string
+				var skips map[string]struct{}
+				if bundleInfo, ok := bundleMap[node.BundlePath]; ok {
```

Comment:
What if one bundle appears in two channels with different skips/replaces? I don't think bundlepath is a precise-enough identifier.

Reply:
We aggregate the skips/replaces across channels, since we can't have the same version twice in the nested bundle format.